pkg/correlation/README.md

```diff
@@ -0,0 +1,103 @@
+## Correlation logging instructions
+
+There are three ways for using this package. All methods will result in a correlation ID being 
+added to the log lines. Each method may make sense in different scenarios. We've provided
+all three methods as they can be used interchangeably depending on the developers preference. 
+
+Currently, a correlation context is created as part of a gRPC interceptor for the following openstorage packages:
+- The CSI Driver (/csi)
+- The SDK Server (/api/server/sdk)
+
+### Method 1: Per-package correlation logger
+This method is nice if a package maintainer wants to register a global 
+logging hook at a per-package level. This is nice as you only need to
+create the logger once per package. However, each log line can be quite
+verbose as you must provide WithContext(ctx)
+
+File 1:
```

Comment:
Suggestion: Always add an empty line before markdown tags

Also you can add the word ```go``` just after the ticks so that github adds Go colors